src/cascadia/TerminalApp/AppLogic.h

```diff
@@ -56,124 +46,47 @@ namespace winrt::TerminalApp::implementation
-        void IdentifyWindow();
-        void RenameFailed();
-        winrt::hstring WindowName();
-        void WindowName(const winrt::hstring& name);
-        uint64_t WindowId();
-        void WindowId(const uint64_t& id);
-        void SetPersistedLayoutIdx(const uint32_t idx);
-        void SetNumberOfOpenWindows(const uint64_t num);
-        bool IsQuakeWindow() const noexcept;
-        void RequestExitFullscreen();
-
-        Windows::Foundation::Size GetLaunchDimensions(uint32_t dpi);
-        bool CenterOnLaunch();
-        TerminalApp::InitialPosition GetInitialPosition(int64_t defaultInitialX, int64_t defaultInitialY);
-        winrt::Windows::UI::Xaml::ElementTheme GetRequestedTheme();
-        Microsoft::Terminal::Settings::Model::LaunchMode GetLaunchMode();
-        bool GetShowTabsInTitlebar();
-        bool GetInitialAlwaysOnTop();
-        float CalcSnappedDimension(const bool widthOrHeight, const float dimension) const;
-
-        Windows::UI::Xaml::UIElement GetRoot() noexcept;
-
-        void SetInboundListener();
-
-        hstring Title();
-        void TitlebarClicked();
-        bool OnDirectKeyEvent(const uint32_t vkey, const uint8_t scanCode, const bool down);
-
-        void CloseWindow(Microsoft::Terminal::Settings::Model::LaunchPosition position);
-        void WindowVisibilityChanged(const bool showOrHide);
-
-        winrt::TerminalApp::TaskbarState TaskbarState();
-        winrt::Windows::UI::Xaml::Media::Brush TitlebarBrush();
-        void WindowActivated(const bool activated);
-
-        bool GetMinimizeToNotificationArea();
-        bool GetAlwaysShowNotificationIcon();
-        bool GetShowTitleInTitlebar();
-
-        winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::UI::Xaml::Controls::ContentDialogResult> ShowDialog(winrt::Windows::UI::Xaml::Controls::ContentDialog dialog);
-        void DismissDialog();
 
         Windows::Foundation::Collections::IMapView<Microsoft::Terminal::Control::KeyChord, Microsoft::Terminal::Settings::Model::Command> GlobalHotkeys();
 
         Microsoft::Terminal::Settings::Model::Theme Theme();
 
-        // -------------------------------- WinRT Events ---------------------------------
-        // PropertyChanged is surprisingly not a typed event, so we'll define that one manually.
-        // Usually we'd just do
-        //    WINRT_CALLBACK(PropertyChanged, Windows::UI::Xaml::Data::PropertyChangedEventHandler);
-        //
-        // But what we're doing here is exposing the Page's PropertyChanged _as
-        // our own event_. It's a FORWARDED_CALLBACK, essentially.
-        winrt::event_token PropertyChanged(Windows::UI::Xaml::Data::PropertyChangedEventHandler const& handler) { return _root->PropertyChanged(handler); }
-        void PropertyChanged(winrt::event_token const& token) { _root->PropertyChanged(token); }
+        TerminalApp::TerminalWindow CreateNewWindow();

```

Comment:
q: does `AppLogic` not keep track of its windows?

Reply:
Sure doesn't! I didn't think that was something it _needed_ to know.